src/Quizzer.API.App/Controllers/AnswerController.cs

```diff
@@ -4,75 +4,81 @@
 
 namespace Quizzer.API.Controllers {
     [ApiController]
-    [Route("api/answers")]
-    public class AnswerController : ControllerBase {
-        private readonly ILogger<AnswerController> _logger;
-        private readonly AnswerFacade _answerFacade;
+    [Route("api/[controller]")]
+    public class AnswerController : ControllerBase{
+        private readonly AnswerFacade _facade;
 
-        public AnswerController(ILogger<AnswerController> logger,
-            AnswerFacade answerFacade) {
-            _logger = logger;
-            _answerFacade = answerFacade;
+        public AnswerController(AnswerFacade facade) {
+            _facade = facade;
         }
 
-        [HttpGet(Name = "GetAll")]
+        [HttpGet(Name = "[controller]/GetAll")]
         public IEnumerable<AnswerDetailModel> GetAll() {
-            return _answerFacade.GetAll();
+            return _facade.GetAll();
         }
 
-        [HttpGet("{id:guid}", Name = "GetById")]
+        [HttpGet("{id:guid}", Name = "[controller]/GetById")]
         [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(Guid))]
         [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(AnswerDetailModel))]
         public IActionResult GetById(Guid id) {
-            AnswerDetailModel answer = _answerFacade.GetDetailById(id);
-            
-            if (answer.Id != id)
+            AnswerDetailModel? answer = _facade.GetDetailById(id);
+
+            if (answer is null)
                 return NotFound(id);
-      
+
             return Ok(answer);
         }
 
-        [HttpGet("byQuestion/{questionId:guid}", Name = "GetForQuestion")]
+        [HttpGet("ByQuestion/{questionId:guid}", Name = "[controller]/GetForQuestion")]
         public IEnumerable<AnswerDetailModel> GetForQuestion(Guid questionId) {
-            return _answerFacade.GetAnswersForQuestion(questionId);
+            return _facade.GetAnswersForQuestion(questionId);
         }
 
-        [HttpPut(Name = "UpdateAnswer")]
+        [HttpDelete("{id:guid}", Name = "[controller]/Delete")]
+        [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(Guid))]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Guid))]
+        public IActionResult Delete(Guid id) {
+            if (!_facade.Exists(id))
+                return NotFound(id);
+
+            _facade.Delete(id, DateTime.Now);
```

Comment:
If the quiz has already started, an exception is thrown. So it means that API wil never notice a "requestor" that operation has failed.

Reply:
updated